### PR TITLE
Allow passing in authentication client secret as an environment variable

### PR DESCRIPTION
**Pull request by katrogan**
# TL;DR
Allow passing in authentication client secret as an environment variable in addition to file path location

## Type
 - [ ] Bug Fix
 - [x] Feature
 - [ ] Plugin

## Are all requirements met?

 - [x] Code completed
 - [x] Smoke tested
 - [ ] Unit tests added
 - [ ] Code documentation added
 - [ ] Any pending items have an associated Issue

## Complete description
 _How did you fix the bug, make the feature etc. Link to any design docs etc_

## Tracking Issue
https://github.com/flyteorg/flyte/issues/2744

## Follow-up issue
_NA_

### CODE REVIEW

File: clients/go/admin/token_source_provider.go [hunk 1]
```diff
@@ -134,14 +135,18 @@ type ClientCredentialsTokenSourceProvider struct {
...
-		logger.Errorf(ctx, "Error reading secret from location %s", cfg.ClientSecretLocation)
-		return nil, err
+	var secret string
+	if len(cfg.ClientSecretLocation) > 0 {
+		secretBytes, err := ioutil.ReadFile(cfg.ClientSecretLocation)
+		if err != nil {
+			logger.Errorf(ctx, "Error reading secret from location %s", cfg.ClientSecretLocation)
+			return nil, err
+		}
+		secret = strings.TrimSpace(string(secretBytes))
```
Comment (pmahindrakar-oss) [inline clients/go/admin/token_source_provider.go:145]: Can we have this done in both the conditions

**Review by pmahindrakar-oss — APPROVED** [review]

File: clients/go/admin/token_source_provider.go (hunk 1 above)
Comment (katrogan) [reply clients/go/admin/token_source_provider.go]: done

**Review by pmahindrakar-oss — APPROVED** [review]

**Review by EngHabu — APPROVED** [review]

**Review by wild-endeavor — APPROVED** [review]
this is just for the CC/api key type auth right?

**Comment by katrogan** [comment]
> this is just for the CC/api key type auth right?
yes (I don't think there's any equivalent for pkce, right?)

**Pull request merged**